routing/payment_lifecycle.go

```diff
@@ -338,12 +338,12 @@ func (p *paymentLifecycle) sendPaymentAttempt(firstHop lnwire.ShortChannelID,
 // handleSendError inspects the given error from the Switch and determines
 // whether we should make another payment attempt.
 func (p *paymentLifecycle) handleSendError(sendErr error) error {
-	var finalOutcome bool
+	var finalOutcome *channeldb.FailureReason
 
 	// If an internal, non-forwarding error occurred, we can stop trying.
 	fErr, ok := sendErr.(*htlcswitch.ForwardingError)
 	if !ok {
-		finalOutcome = true
+		finalOutcome = channeldb.FailureReasonError.Ref()
```

Comment:
Hmm, will the compiler allow one to do `*finalOutcome = somethingElse=? Given that you're obtaining a pointer to a constant.

Reply:
It is not a pointer to a constant. The `Ref()` function creates a new variable with the same value as the const and returns a pointer to that variable. Pointer to constant if impossible afaik.

Reply:
Changed `processSendError` to return a `final` bool and the reason. On second thought the `Ref()` thing was a bit too weird.